buildSrc/subprojects/jvm/src/main/kotlin/gradlebuild/propagated-env-variables.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gradlebuild
+
+import org.gradle.api.tasks.testing.Test
+import gradlebuild.basics.BuildEnvironment
+
+val propagatedEnvAllowList = listOf(
+    "OS",
+    "CLASSPATH",
+    "ANDROID_HOME",
+    "DIRNAME",
+    "USERNAME",
+    "USERDOMAIN",
+    "REPO_MIRROR_URL",
+    "APP_HOME",
+    "TMPDIR",
+    "TMP",
+    "TEMP"
+)
+
+fun Test.configurePropagatedEnvVariables() {
+    if (BuildEnvironment.isCiServer) {
+        environment(System.getenv().entries.map(::sanitize).toMap())
```

Comment:
How about filtering the entries instead of replacing the non-allowed entries by an empty String?

Reply:
IIUC, we have to overwrite them with empty string.

Let's say the main build daemon has env variable `USERNAME=user`/`PASSWORD=myPassword`, if we configure the test JVMs with only `USERNAME=1`, then `PASSWORD=myPassword` is still propagated to forked processes automatically. We have to pass `USERNAME=user`/`PASSWORD=""` to avoid the sensitive env variables leaking.

Reply:
Yes, that is true. What happens if you use `Test.setEnvironment` instead of `environment`?

Reply:
I think if we overwrite the environment variables, then `setEnvironment` and `environment` are no different.

Reply:
That is true. Though I think with `setEnvironment` we don't even pass the empty environment variables any more. I would prefer that to passing empty environment variables.